picoquic/logger.c

```diff
@@ -1811,73 +1811,45 @@ void picoquic_log_picotls_ticket(FILE* F, picoquic_connection_id_t cnx_id,
 
 void picoquic_open_cc_dump(picoquic_cnx_t * cnx)
 {
-    int ret = 0;
-
     if (cnx->cc_log != NULL) {
         DBG_PRINTF("%s", "CC LOG File is already open!\n");
-    } else if (cnx->quic->cc_log_dir != NULL) {
-        char cc_log_file_name[512];
-        char const * suffix = "-log.bin";
-        const size_t suffix_len = strlen(suffix);
-        size_t folder_length = strlen(cnx->quic->cc_log_dir);
-
-        if (folder_length + 1u + (size_t)cnx->initial_cnxid.id_len*2u + suffix_len + 1u > sizeof(cc_log_file_name)) {
-            ret = -1;
-        }
-        else {
-            size_t byte_index = 0;
-
-            memcpy(cc_log_file_name, cnx->quic->cc_log_dir, folder_length);
-            byte_index += folder_length;
-#ifdef _WINDOWS
-            cc_log_file_name[byte_index++] = '\\';
-#else
-            cc_log_file_name[byte_index++] = '/';
-#endif
-            for (size_t i = 0; ret == 0 && i < cnx->initial_cnxid.id_len; i++) {
-#ifdef _WINDOWS
-                ret = sprintf_s(&cc_log_file_name[byte_index], sizeof(cc_log_file_name) - byte_index, "%02x", cnx->initial_cnxid.id[i]) <= 0;
-#else
-                ret = sprintf(&cc_log_file_name[byte_index], "%02x", cnx->initial_cnxid.id[i]) <= 0;
-#endif
-
-                byte_index += 2;
-            }
+        return;
```

Comment:
I could not find where the deleted code went. It is actually used on the server side, where we want one cc log file per connection. The convention is to name that file based on the initial CID of the connection. This will then enable retrieval of the trace by the tester, using something similar to the support for connection logs at https://test.privateoctopus.com/.

Reply:
The deleted code mostly went into the single sprintf call now.

Reply:
I don't understand. There should be a branch here thatgoes something like:
~~~
if file is already open
    do nothing
else
    format file name in selected directory using sprintf
    open file
~~~
I don't see that code anymore.

Reply:
I didn't change that. That's why it is not in the diff, but it is still in the code.

Reply:
See the full file here:
https://github.com/steschu77/picoquic/blob/refactor_open_cc_log/picoquic/logger.c#L1812

Reply:
OK. I was not seeing the complete picture.